mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/NullSavingStrategyValidation.java

```diff
@@ -0,0 +1,246 @@
+            && nullSavingStrategyExplicitlySet(v, SetEntity.class);
+      } else {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+      }
+    }
+    return false;
+  }
+
+  private boolean insertHasDoNotSet(ExecutableElement v, boolean explicitSet) {
+    Insert annotation = v.getAnnotation(Insert.class);
+    if (annotation != null) {
+      if (explicitSet) {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET
+            && nullSavingStrategyExplicitlySet(v, Insert.class);
+      } else {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+      }
+    } else {
+      return false;
+    }
+  }
+
+  private boolean updateHasDoNotSet(ExecutableElement v, boolean explicitSet) {
+    Update annotation = v.getAnnotation(Update.class);
+    if (annotation != null) {
+      if (explicitSet) {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET
+            && nullSavingStrategyExplicitlySet(v, Update.class);
+      } else {
+        return annotation.nullSavingStrategy() == NullSavingStrategy.DO_NOT_SET;
+      }
+    } else {
+      return false;
+    }
+  }
+
+  private boolean queryHasSetToNullExplicitly(ExecutableElement v) {
+    Query annotation = v.getAnnotation(Query.class);
+    if (annotation != null) {
+      return annotation.nullSavingStrategy() == NullSavingStrategy.SET_TO_NULL
+          && nullSavingStrategyExplicitlySet(v, Query.class);
+    }
+    return false;
+  }
+
+  private boolean setEntitySetToNullExplicitly(ExecutableElement v) {
+    SetEntity annotation = v.getAnnotation(SetEntity.class);
+    if (annotation != null) {
+      return annotation.nullSavingStrategy() == NullSavingStrategy.SET_TO_NULL
+          && nullSavingStrategyExplicitlySet(v, SetEntity.class);
+    }
+    return false;
+  }
+
+  private boolean insertHasSetToNullExplicitly(ExecutableElement v) {
+    Insert annotation = v.getAnnotation(Insert.class);
+    if (annotation != null) {
+      return annotation.nullSavingStrategy() == NullSavingStrategy.SET_TO_NULL
+          && nullSavingStrategyExplicitlySet(v, Insert.class);
+    } else {
+      return false;
+    }
+  }
+
+  private boolean updateHasSetToNullExplicitly(ExecutableElement v) {
+    Update annotation = v.getAnnotation(Update.class);
+    if (annotation != null) {
+      return annotation.nullSavingStrategy() == NullSavingStrategy.SET_TO_NULL
+          && nullSavingStrategyExplicitlySet(v, Update.class);
+    } else {
+      return false;
+    }
+  }
+
+  private boolean isOperationWithNullSavingStrategy(ExecutableElement v) {
+    return v.getAnnotation(Update.class) != null
+        || v.getAnnotation(Insert.class) != null
+        || v.getAnnotation(SetEntity.class) != null
+        || v.getAnnotation(Query.class) != null;
+  }
+
+  boolean nullSavingStrategyExplicitlySet(Element methodElement, Class<?> javaClass) {
```

Comment:
If we were to remove `default NullSavingStrategy.DO_NOT_SET` from all annotations so they are null instead couldn't this be more effective as all you would have to do is null check?

Reply:
Removing the default makes the attribute is mandatory on the annotation.

Reply:
ah, right.  That didn't occur to me at the time, in any case, could we just set default to null?

Reply:
If we will set it to `null` we will still need to interpret is somehow for a case where 
`@DefaultNullSavingStrategy` on the `@Dao` level is not a set AND the method level `nullSavingStrategy` property for example on `@Update` will not be set. 
In such case that null needs to become `DO_NOT_SET` or `SET_TO_NULL`. I think that it is better to have that default logic embedded within the annotation default:
`NullSavingStrategy nullSavingStrategy() default NullSavingStrategy.DO_NOT_SET;`
instead of adding another case to handle when the value is null.

Reply:
Plus `null` is not a valid default for an annotation parameter.